pkg/stdcopy/stdcopy.go

```diff
@@ -155,11 +160,18 @@ func StdCopy(dstout, dsterr io.Writer, src io.Reader) (written int64, err error)
 			}
 		}
 
+		// we might have an error from the daemon mixed up in our multiplexed http
+		// stream. if we do, return it.
+		if out == nil {
+			return written, fmt.Errorf("got error from daemon: %v", string(buf[stdWriterPrefixLen:frameSize+stdWriterPrefixLen]))
```

Comment:
Things only enter the Systemerr stream if ContainerLogs returns an error, so the error will always be at the end of the stream anyway. I don't think it's necessary to spend the extra effort of fixing it to return the error after writing.